signwriting_animation/diffusion/tests/test_overfitting.py

```diff
@@ -0,0 +1,118 @@
+import torch
+import torch.optim as optim
+import matplotlib.pyplot as plt
+from transformers import CLIPProcessor
+
+from signwriting_animation.diffusion.core.models import SignWritingToPoseDiffusion
+from signwriting_evaluation.metrics.clip import signwriting_to_clip_image
+
+# Set random seed for reproducibility
+torch.manual_seed(42)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed_all(42)
+
+def make_sample(signwriting_str, value, device, clip_processor,
+                batch_size=1, num_past_frames=10, num_keypoints=21, num_dims=3):
+    """
+    Create a dummy sample: all-zero or all-one pose, with SignWriting image embedding.
+    """
+    x = torch.full((batch_size, num_keypoints, num_dims, num_past_frames), value, device=device)
+    timesteps = torch.zeros(batch_size, dtype=torch.long, device=device)
+    past_motion = torch.full((batch_size, num_keypoints, num_dims, num_past_frames), value, device=device)
+    pil_img = signwriting_to_clip_image(signwriting_str)
+    sw_img = clip_processor(images=pil_img, return_tensors="pt").pixel_values.to(device)
+    return x, timesteps, past_motion, sw_img
+
+def main():
```

Comment:
this file should be a proper test file - that means, the functions should be `test_model_overfits` and it should run using pytest.

See this overfit test file for example:
https://github.com/sign-language-processing/transcription/blob/main/text_to_pose/diffusion/tests/model_overfit_test.py#L26